src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -916,11 +967,7 @@ private async Task AutoCloseOnPipesClosureAsync()
 
             private void Fault(Exception exception)
             {
-                if (this.TraceSource?.Switch.ShouldTrace(TraceEventType.Critical) ?? false)
-                {
-                    this.TraceSource!.TraceEvent(TraceEventType.Critical, (int)TraceEventId.FatalError, "Channel Closing self due to exception: {0}", exception);
-                }
-
+                // If the reason why are faulting isn't already set then do so before disposing the channel
```

Comment:
I was actually experimenting with only show a couple of trace messages that I cared about and somehow this removal got swept into these changes. I will immediately revert this change because that should definitely not have been removed